File: Ch 1 Array and Strings/1_6_stringCompression/1_6_stringCompression.py
import sys

def compress(old_str):
	new_str = ""
	print "old str:", old_str
	i = 0
	while i < len(old_str):
		j = i+1
		count = 1
		while j < len(old_str) and old_str[j] == old_str[i]:
			j += 1
			count += 1
		new_str += old_str[i] + str(count)
		i = j
	print "new str:", new_str
	if len(new_str) >= len(old_str):
		print "do not compress"

if __name__ == '__main__':
	args = sys.argv
	if len(args) == 2:
		compress(args[1])
	else:
		compress("hellowolrd")
	# compress

File: Ch 1 Array and Strings/1_6_stringCompression/1_6_stringCompression.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cstdlib>

using namespace std;

void compress(string & str){
	cout << "old string: " << str << endl;
	string new_str = "";
	int i=0;
	while(i<str.length()){
		char current = str[i];
		int count = 1;
		int j = i+1;
		while(j < str.length() && str[j] == current){
			count++;
			j++;
		}
		new_str += current;
		// problem with using std:to_stting or itoa, wrote solution in python instead
		char * intstr = itoa(count);
		new_str += string(intstr);
		i = j;

	}

	cout << "new string: " << new_str << endl;
	if(new_str.length() >= str.length()){
		cout << "do not convert"<<endl;
	}

	/*
	for(int i=0; i<str.length(); i++){
		char current = str[i];
		int count = 1;
		for(int j=i+1; j<str.length();j++){
			if(str[j] == current){
				count++;
			}
			else{
				break;
			}
		}
		i = 
	}
	*/

}

int main(){
	string str;
	cin >> str;
	compress(str);
}